requirements.txt

```diff
@@ -14,15 +14,13 @@ keyring==24.2.0
 LabJackPython==2.1.0
 lazy_loader==0.2
 lxml==4.9.2
-matplotlib==3.7.1
 mypy-extensions==1.0.0
-networkx==3.1
-numpy==1.24.3
+numpy
```

Comment:
Specifying exact version makes creating virtual environments using [`run_in_venv`](https://github.com/flojoy-ai/python/blob/128090e873d87841aff83c2e0b3fa0977bf4d4f0/flojoy/flojoy_node_venv.py#L248) hardly possible since [its virtual environments depends on `flojoy`](https://github.com/flojoy-ai/python/blob/128090e873d87841aff83c2e0b3fa0977bf4d4f0/flojoy/flojoy_node_venv.py#L284) which is itself quite constrained, [especially on the version of pandas and numpy](https://github.com/flojoy-ai/python/blob/main/pyproject.toml#L16-L17).

Reply:
See for instance [the CI run error](https://github.com/flojoy-ai/nodes/actions/runs/6034554658/job/16373157925?pr=268#step:5:245) before 5b0fabb was introduced.

Reply:
I see, makes sense, looks good then!

Reply:
@jjerphan @JeffDotPng The current dep management is a bit messy, we will have to rethink it a bit. 
I created a project on Linear over here: https://linear.app/flojoy/project/python-dependency-hell-565c5991d2ed
Feel free to leave more information there so we can keep track of the status on one place :)